src/main/java/org/reactome/release/resourcechecker/ResourceCheckerFactory.java

```diff
@@ -0,0 +1,31 @@
+package org.reactome.release.resourcechecker;
+
+import org.reactome.release.Resource;
+
+public class ResourceCheckerFactory {
+	public static ResourceChecker getInstance(Resource resource) {
+		switch(resource.getResourceType()) {
+			case FILE:
+				if (resource.getResourceURL().getProtocol().equals("ftp")) {
+					return new FTPFileResourceChecker(resource);
+				} else if (resource.getResourceURL().getProtocol().startsWith("http")) {
+					return new HTTPFileResourceChecker(resource);
+				} else {
+					throw new IllegalArgumentException(
+						"The protocol " + resource.getResourceURL().getProtocol() +
+						" to check a file resource is not supported for " + resource.getResourceName() + "\n"
+					);
+				}
+			case REST_ENDPOINT:
+				return new RESTfulAPIResourceChecker(resource);
+			case WEB_PAGE:
+				return new WebPageResourceChecker(resource);
+			case WEB_SERVICE:
```

Comment:
Why is there no action for WEB_SERVICE?

Reply:
For FTP server, it's handled as one of the conditions in `case FILE:` and for `case WEB_SERVICE:`, it's a work in progress.  I may change the resource type enum so `FILE` becomes `FTP_FILE` and `HTTP_FILE` and they get their own cases here.